File: java/src/main/java/com/android/frame/mvp/extra/RecyclerView/LoadMoreAdapter.java
package com.android.frame.mvp.extra.RecyclerView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.module.LoadMoreModule;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * Created by xuzhb on 2020/12/31
 * Desc:上拉加载更多
 */
public abstract class LoadMoreAdapter<T> extends BaseQuickAdapter<T, BaseViewHolder> implements LoadMoreModule {

    public LoadMoreAdapter(int layoutResId) {
        super(layoutResId);
    }

    public LoadMoreAdapter(int layoutResId, @Nullable List<T> data) {
        super(layoutResId, data);
    }

}

File: kotlin/src/main/java/com/android/frame/mvc/extra/RecyclerView/LoadMoreAdapter.kt
package com.android.frame.mvc.extra.RecyclerView

import androidx.annotation.LayoutRes
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.module.LoadMoreModule
import com.chad.library.adapter.base.viewholder.BaseViewHolder

/**
 * Created by xuzhb on 2020/7/28
 * Desc:上拉加载更多
 */
abstract class LoadMoreAdapter<T> @JvmOverloads constructor(
    @LayoutRes private val layoutResId: Int,
    data: MutableList<T>? = null
) : BaseQuickAdapter<T, BaseViewHolder>(layoutResId, data), LoadMoreModule